puppet-client.html

```diff
@@ -60,18 +61,122 @@
+                box-shadow: inset 0px 1px 0px 0px #ffffff;
+                background: linear-gradient(to bottom, #f9f9f9 5%, #e9e9e9 100%);
+                background-color: #f9f9f9;
+                border-radius: 2px;
+                border: 1px solid #dcdcdc;
+                display: inline-block;
+                cursor: pointer;
+                color: #666666;
+                padding: 2px 10px;
+                text-decoration: none;
+                text-shadow: 0px 1px 0px #ffffff;
+            }
+
+            .error-toolbar__button--main {
+                font-weight: bold;
+            }
+
+            .error-toolbar__button:hover {
+                background: linear-gradient(to bottom, #e9e9e9 5%, #f9f9f9 100%);
+                background-color: #e9e9e9;
+            }
+
+            .error-toolbar__button:active {
+                position: relative;
+                top: 1px;
             }
         </style>
-        <div class="box-container" hidden$="{{!showReconnectingIn}}"><div class="box notice"><strong>Not connected.</strong> Connecting in {{reconnectionSeconds}}s... <a href="" on-tap="reconnectNow">Retry now</a></div></div>
-        <div class="box-container" hidden$="{{!showReconnectingNow}}"><div class="box notice"><strong>Connecting now...</strong></div></div>
-        <div class="box-container" hidden$="{{!connectionError.length}}"><div class="box error"><strong>Connection error:</strong> <pre>{{connectionError}}</pre><a href="" on-tap="reload">Reload page</a> </div></div>
+        <div class="box-container" hidden$="{{!showReconnectingIn}}">
+            <div class="box notice">
+                <div class="error-header error-header--centered"><strong>Not connected.</strong> Connecting in {{reconnectionSeconds}}s... <a href="" on-tap="reconnectNow">Retry now</a></div>
+            </div>
+        </div>
+        <div class="box-container" hidden$="{{!showReconnectingNow}}">
+            <div class="box notice">
+                <div class="error-header error-header--centered"><strong>Connecting now...</strong></div>
+            </div>
+        </div>
+        <div class="box-container" hidden$="{{!connectionError.raw}}">
+            <div class="box error">
+                <div class="error-header error-header--centered">
+                    <strong>Connection error</strong>
+                </div>
+
+                <div class="error-detail">
+                    <div class="error-detail__item">
+                        <template is="dom-if" if="{{connectionError.method}}">
+                            <strong>{{connectionError.method}}</strong>
+                        </template>
+                        <template is="dom-if" if="{{connectionError.url}}">
+                            {{connectionError.url}}
+                        </template>
+                    </div>
+
+                    <template is="dom-if" if="{{connectionError.data.reason}}">
+                        <!-- WS -->
```

Comment:
I think that puppet-client shouldn't distinguish between this errors here. Instead of this, puppet itself could pack error messages into, say, "description" and "raw" fields.